### PR TITLE
Add new addPlaceServlet

### PR DESCRIPTION
**Pull request by sophia-qin**
Allows user to input location name, x coordinate, and y coordinate.

### CODE REVIEW

File: frontend/src/addPlace.js [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+    this.state = { name: "", x_coord: "", y_coord: "" };
+  }
+
+  render() {
+    const { name, x_coord, y_coord } = this.state;
+    const placeObject = { name, x_coord, y_coord };
+
+    return (
+      <div>
+        <h1> Add Place </h1>
```
Comment (jajanet) [inline frontend/src/addPlace.js]: May remove extra space here?

File: src/main/java/com/google/codeu/servlets/AddPlaceServlet.java [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+
+@WebServlet("/addPlace")
+public class AddPlaceServlet extends HttpServlet{
+    @Override
+    public void doGet(HttpServletRequest request, HttpServletResponse response)
+            throws IOException, ServletException {
+        Properties props = ServletUtils.GetProperties();
+        request.setAttribute("TITLE", "Add Place");
+        request.setAttribute(
+                "HEAD_HTML", "<link rel='stylesheet' href='/css/main.css'>"
```
[the remark this answers is not in the document]
Comment (jajanet) [reply src/main/java/com/google/codeu/servlets/AddPlaceServlet.java]: yeah i think once i add my CSS she would need it

File: src/main/java/com/google/codeu/servlets/BlobstoreServlet.java [hunk 3]
```diff
@@ -0,0 +1,94 @@
...
+import java.util.Map;
+import javax.servlet.ServletOutputStream;
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import org.jsoup.Jsoup;
+import org.jsoup.safety.Whitelist;
+
+@WebServlet(("/my-form-handler"))
```
Comment (jajanet) [inline src/main/java/com/google/codeu/servlets/BlobstoreServlet.java]: double parentheses, maybe remove a pair

File: src/main/java/com/google/codeu/servlets/MessageServlet.java [hunk 4]
```diff
@@ -79,13 +89,70 @@ public void doPost(HttpServletRequest request, HttpServletResponse response)
     }
 
     String user = userService.getCurrentUser().getEmail();
+    // Get the message entered by the user.
     String userText =
         Jsoup.clean(request.getParameter("text"), Whitelist.none());
-    String replacement = "<img src=\"$1\" />";
-    String textWithImagesReplaced = userText.replaceAll(REGEX, replacement);
+    /*   String replacement = "<img src=\"$1\" />";
```
Comment (jajanet) [inline src/main/java/com/google/codeu/servlets/MessageServlet.java]: should this be commented out?? (if so, plz ignore sorry lmaooo)

File: src/main/java/com/google/codeu/servlets/AddPlaceServlet.java (hunk 2 above)
[the remark this answers is not in the document]
Comment (sophia-qin) [reply src/main/java/com/google/codeu/servlets/AddPlaceServlet.java]: done

File: frontend/src/addPlace.js [hunk 5]
```diff
@@ -0,0 +1,42 @@
...
+        <textarea>Enter a location name</textarea>
```
[the remark this answers is not in the document]
Comment (sophia-qin) [reply frontend/src/addPlace.js]: fixed

File: src/main/java/com/google/codeu/servlets/MessageServlet.java (hunk 4 above)
Comment (sophia-qin) [reply src/main/java/com/google/codeu/servlets/MessageServlet.java]: took it out!

File: src/main/java/com/google/codeu/servlets/BlobstoreServlet.java (hunk 3 above)
Comment (sophia-qin) [reply src/main/java/com/google/codeu/servlets/BlobstoreServlet.java]: removed!

File: frontend/src/addPlace.js (hunk 1 above)
Comment (sophia-qin) [reply frontend/src/addPlace.js]: removed!

File: frontend/src/addPlace.js [hunk 6]
```diff
@@ -0,0 +1,43 @@
...
+
+  render() {
+    const { name, x_coord, y_coord } = this.state;
+    const placeObject = { name, x_coord, y_coord };
+
+    return (
+      <div>
+        <h1>Add Place</h1>
+        <textarea value={this.state.value} onChange={e => {this.setState({ name: e.target.value})}}>
+        Enter a location name</textarea>
```
Comment (tylerhou) [inline frontend/src/addPlace.js]: Textarea shouldn't have any children

File: frontend/src/addPlace.js [hunk 7]
```diff
@@ -0,0 +1,43 @@
...
+        <textarea value={this.state.value} onChange={e => {this.setState({ y_coord: e.target.value })}}>Enter a Y coord</textarea>
+        <br />
+        <button
+          onClick={() =>
+            fetch("/api/place", {
+              method: "POST",
+              credentials: "same-origin",
+              headers: { "Content-Type": "application/json" },
+              body: JSON.stringify(placeObject)
+            }).then(() => window.location.reload())
```
Comment (tylerhou) [inline frontend/src/addPlace.js]: Why do we need to reload?

**Pull request closed without merging**